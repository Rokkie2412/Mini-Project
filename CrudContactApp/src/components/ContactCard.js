/**
 * @format
 * @flow
 */
import * as React from 'react'
import type {Cards} from './component.type'
import {useState} from 'react'
import {View,KeyboardAvoidingView,Pressable,Text,Image} from 'react-native'
import Modal from "react-native-modal";
import styles from '../styles/ContactCard.Style'
import Icon from 'react-native-vector-icons/Ionicons'

const ContactCard = ({show,setshow,image,first,last,Age}:Cards):React.Node =>{
    console.log(image)
    return(
        <Modal
        isVisible={show}
        style={styles.mainContainer}
        >
            <Text style={styles.Header}>Contact Card</Text>
            {image === "N/A" ? <Icon style={styles.personIcon} name='person'/>:
            <Image style={styles.imagePerson} source={{uri:`${image}`}}/>
            }
            <View style={styles.name}>
                <Text style={styles.text}>First Name:  </Text>
                <Text style={styles.textName}>{first}</Text>
            </View>
            <View style={styles.name}>
                <Text style={styles.text}>Last Name:  </Text>
                <Text style={styles.textName}>{last}</Text>
            </View>
            <View style={styles.name}>
                <Text style={styles.text}>Age:  </Text>
                <Text style={styles.textName}>{Age}</Text>
            </View>
            <View style={styles.button}>
                <Pressable
                onPress={()=>{
                    setshow(!show)
                }}
                >
                    <Text style={styles.editButton}>Close</Text>
                </Pressable>
                <Pressable>
                    <Text style={styles.editButton}> Edit </Text>
                </Pressable>
                
            </View>
        </Modal>
    )
}

export default ContactCard