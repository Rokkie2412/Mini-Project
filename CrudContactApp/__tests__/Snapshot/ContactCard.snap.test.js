import {create,act} from 'react-test-renderer';
import React from 'react';
import ContactCard from "../../src/components/ContactCard";

jest.mock('react-native-modal',()=> "Mocked Modal React Native")
jest.mock('react-native-vector-icons/Ionicons', () => 'Mocked Ionicons');


const func = jest.fn()
const tree = create(<ContactCard setshow={func} image={"N/A"}/>)

describe('Testing for Contact Card', () => {
  describe('Snapshot for Contact Card', () => {
    it('Should be render correctly',()=>{
        expect(tree).toMatchSnapshot()
    })
  })
  describe('Button test for Contact Card', () => {
    it('Should be pressed',()=>{
        const button = tree.root.findByProps({testID:'cancel'}).props
        act(()=>button.onPress())
    })
  })
  
})
