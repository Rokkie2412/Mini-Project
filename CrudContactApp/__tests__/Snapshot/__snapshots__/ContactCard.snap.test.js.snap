// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing for Contact Card Snapshot for Contact Card Should be render correctly 1`] = `
<Mocked Modal React Native
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "#393E46",
      "borderRadius": 30,
      "flex": 1,
      "marginVertical": "50%",
      "position": "absolute",
      "width": "70%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#EEEEEE",
        "fontSize": 24,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    Contact Card
  </Text>
  <Mocked Ionicons
    name="person"
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#00ADB5",
        "borderRadius": 50,
        "color": "#EEEEEE",
        "fontSize": 50,
        "margin": "9%",
        "padding": 15,
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "center",
        "flexDirection": "row",
        "width": "40%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#EEEEEE",
          "fontWeight": "bold",
        }
      }
    >
      First Name:  
    </Text>
    <Text
      style={
        Object {
          "borderBottomWidth": 1,
          "borderColor": "#00ADB5",
          "color": "#EEEEEE",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "flexDirection": "row",
        "width": "40%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#EEEEEE",
          "fontWeight": "bold",
        }
      }
    >
      Last Name:  
    </Text>
    <Text
      style={
        Object {
          "borderBottomWidth": 1,
          "borderColor": "#00ADB5",
          "color": "#EEEEEE",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "flexDirection": "row",
        "width": "40%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#EEEEEE",
          "fontWeight": "bold",
        }
      }
    >
      Age:  
    </Text>
    <Text
      style={
        Object {
          "borderBottomWidth": 1,
          "borderColor": "#00ADB5",
          "color": "#EEEEEE",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-evenly",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="cancel"
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#00ADB5",
            "borderRadius": 30,
            "color": "#EEEEEE",
            "fontSize": 16,
            "fontWeight": "bold",
            "marginTop": 20,
            "padding": 12,
          }
        }
      >
        Close
      </Text>
    </View>
  </View>
</Mocked Modal React Native>
`;
